velox/docs/develop/types.rst

```diff
@@ -73,6 +73,28 @@ Examples:
 * Timestamp(-5000*24*60*60 - 1000, 123456) represents 1956-04-24 07:43:20.000123456
   (5000 days 1000 seconds before epoch plus 123456 nanoseconds).
 
+Floating point types (float, double) have special values negative/positive infinity and
+not-a-number (NaN).
+For negative/positive infinity the following standard floating point semantics apply:
+
+* Positive infinity multiplied by any positive value returns positive infinity.
+* Negative infinity multiplied by any positive value returns negative infinity.
+* Positive infinity multiplied by any negative value returns negative infinity.
+* Negative infinity multiplied by any negative value returns positive infinity.
+* Positive/negative infinity multiplied by 0 returns NaN.
+* Positive/negative infinity is equal to itself.
+* In aggregations, all positive infinity values are grouped together. Similarly, all negative infinity values are grouped together.
+* Positive infinity and negative infinity are treated as normal values in join keys.
+* Positive infinity sorts lower than NaN and higher than any other values.
+* Negative infinity sorts lower than any other values.
+
+For NaN the semantics are different than the standard floating point semantics:
+
+* NaN = NaN returns true.
+* In aggregations, all NaN values are grouped together.
+* NaN is treated as a normal value in join keys.
+* NaN values go last when in ascending order, larger than any other numeric value.
```

Comment:
Any special semantics between NaN, +/- Infinity, and NULLs?

Reply:
No. NaN, +/- Infinity are treated as values, NULL is not a value. NULL is treated separately - notably the ordering has options if NULLs are returned first or last. The rest of the non-NULL values are ordered as per the semantics described. That was one of my PRs where float/double is part of complex types. The comparison implementation was different than for the simple types. The tests specifically test the ordering of the special values and NULL.